spec/javascripts/health-care/components/questions/FullName.spec.jsx

```diff
@@ -1,80 +1,54 @@
 import React from 'react';
 import ReactTestUtils from 'react-addons-test-utils';
+import SkinDeep from 'skin-deep';
 
 import FullName from '../../../../../_health-care/_js/components/questions/FullName';
 
 describe('<FullName>', () => {
-  let component = null;
+  describe('propTypes', () => {
+    let consoleStub;
+    beforeEach(() => {
+      consoleStub = sinon.stub(console, 'error');
+    });
+
+    afterEach(() => {
+      consoleStub.restore();
+    });
+
+    // TODO(awong): consider implementing higher-order component approach using invariant
+    xit('value is required', () => {
+      SkinDeep.shallowRender(<FullName/>);
+      sinon.assert.calledWithMatch(consoleStub, /Required prop `value` was not specified in `FullName`/);
+    });
+
+    it('value must be an object', () => {
+      SkinDeep.shallowRender(<FullName value/>);
+      sinon.assert.calledWithMatch(consoleStub, /Invalid prop `value` of type `boolean` supplied to `FullName`, expected `object`/);
+    });
+
+    // TODO(awong): Why in the world does this not work?!?!
+    xit('onValueChange is required', () => {
+      SkinDeep.shallowRender(<FullName/>);
+      sinon.assert.calledWithMatch(consoleStub, /Required prop `onValueChange` was not specified in `FullName`/);
+    });
+
+    xit('onValueChange must be a function', () => {
+      SkinDeep.shallowRender(
+        <FullName onValueChange/>);
+      sinon.assert.calledWithMatch(consoleStub, /Invalid prop `onValueChange` of type `boolean` supplied to `FullName`, expected `function`/);
+    });
+  });
 
-  beforeEach(() => {
-    component = ReactTestUtils.renderIntoDocument(
-      <FullName name={{ first: 'William', last: 'Shakespeare' }} onUserInput={(_update) => {}}/>
+  it('has sane looking features', () => {
+    const component = ReactTestUtils.renderIntoDocument(
+      <FullName value={{ first: '', last: '' }}/>
     );
     assert.ok(component, 'Cannot even render component');
-  });
 
-  it('has sane looking features', () => {
     const inputs = ReactTestUtils.scryRenderedDOMComponentsWithTag(component, 'input');
     expect(inputs).to.have.length(3);
 
     const selects = ReactTestUtils.scryRenderedDOMComponentsWithTag(component, 'select');
     expect(selects).to.have.length(1);
   });
```

Comment:
We should have a SkinDeep baseed integration test here for invalid `value`s like `value = {{first:'$#', middle: '', last: 'lastname', suffix: '' }}` (do one for each field) to ensure validation logic is called per field that we expect validations for.

In the current code, if you wrote these 3 test cases, the one for middle should fail since we have no validation logic on it.